plugin.video.ahltv/resources/lib/ahl_tv.py

```diff
@@ -0,0 +1,253 @@
+
+        if "games" in json_source:
+            log("num games returned: " + str(len(json_source['games'])))
+            try:
+                if len(json_source['games']) > 0:
+                    for game in json_source['games']:
+                        create_game_listitem(game, game_day)
+            except Exception as e:
+                log('Failed to process games: '+ str(e), True)
+                pass
+
+        next_day = display_day + timedelta(days=1)
+        add_dir('[B]%s >>[/B]' % LOCAL_STRING(32004), 101, AHL_LOGO, FANART, next_day.strftime("%Y-%m-%d"))
+
+def create_game_listitem(game, game_day):
+    icon = None
+    home_team = game['homeTeam']
+    visiting_team = game['visitingTeam']
+
+    icon = get_game_icon(home_team['iteamid'], home_team['strteamlogoimageurl'], visiting_team['iteamid'], visiting_team['strteamlogoimageurl'])
+
+    game_time = ''
+    if game['blivegameflag'] == 2 and game['currentlyStreaming'] == False:
+        game_time = game['dateTime']
+        game_time = string_to_date(game_time, "%Y-%m-%dT%H:%M:%SZ")
+        game_time = utc_to_local(game_time)
+        game_time = game_time.strftime('%I:%M %p').lstrip('0')
+
+    elif game['blivegameflag'] == 2 and game['currentlyStreaming'] == True:
+        game_time = '(Live)'
+    else:
+        game_time = '(Final)'
+
+    game_id = game['id']
+
+    fanart = TEAM_FANART[home_team['iteamid']]
+    name = '%s %s %s %s' % (game_time, visiting_team['strteamname'], LOCAL_STRING(32028), home_team['strteamname'])
+    desc = '%s %s %s' % (visiting_team['strteamname'], LOCAL_STRING(32028), home_team['strteamname']) + ' - ' + game_day
+    name = name.encode('utf-8')
+
+    title = '%s %s %s' % (visiting_team['strteamname'], LOCAL_STRING(32028), home_team['strteamname'])
+    title = title.encode('utf-8')
+
+    # Label free game of the day
+    try:
+        if bool(game['free']):
+            name = name + color_string(' (FREE)', COLOR_FREE)
+    except:
+        pass
+
+    video_info = {}
+    audio_info = {}
+
+    info = {'plot': desc,
+            'tvshowtitle': 'AHLTV',
+            'title': title,
+            'aired': game_day,
+            'genre': 'Hockey'}
+
+    add_game_listitem(name, title, game_id, icon, fanart, info, video_info, audio_info)
+
+def color_string(string, color):
+    return '[COLOR=' + color + ']' + string + '[/COLOR]'
+
+def add_game_listitem(name, title, game_id, icon=None, fanart=None, info=None, video_info=None, audio_info=None):
+    ok = True
+
+    u = sys.argv[0] + "?url=&mode=" + str(104) + "&name=" + quote(name) + "&game_id=" + quote(str(game_id))
+
+    liz = xbmcgui.ListItem(name)
+
+    if icon is not None:
+        liz.setArt({'icon': icon, 'thumb': icon, 'fanart': fanart, 'clearlogo': CLEARLOGO})
+    else:
+        liz.setArt({'icon': ICON, 'thumb': ICON, 'fanart': fanart, 'clearlogo': CLEARLOGO})
+
+    if fanart is not None:
```

Comment:
Not needed, already set above. Please drop.

Reply:
I have removed the set of fanart in the top section since it should not be set until the check of `if fanart is not None:`